aqt/archives.py

```diff
@@ -37,6 +51,294 @@ def __init__(self, version, target, arch, os_name):
         self.os_name = os_name
 
 
+class ListCommand:
+    """Encapsulate all parts of the `aqt list` command"""
+
+    def __init__(
+        self,
+        archive_id: ArchiveId,
+        *,
+        filter_minor: Optional[int] = None,
+        is_latest_version: bool = False,
+        modules_ver: Optional[str] = None,
+        extensions_ver: Optional[str] = None,
+        architectures_ver: Optional[str] = None,
+        http_fetcher: Optional[Callable[[str], str]] = None,
+    ):
+        """
+        Construct ListCommand.
+        @param filter_minor         When set, the ListCommand will filter out all versions of
+                                    Qt that don't match this minor version.
+        @param is_latest_version    When True, the ListCommand will find all versions of Qt
+                                    matching filters, and only print the most recent version
+        @param modules_ver          Version of Qt for which to list modules
+        @param extensions_ver       Version of Qt for which to list extensions
+        @param architectures_ver    Version of Qt for which to list architectures
+        @param http_fetcher         Function used to fetch documents via http
```

Comment:
http_fetcher parameter seems to be used only for test purpose, not from Cli class.

I like  a way to override something ListCommand.fetcher() method using pytest monkeypatch.

Reply:
This is correct; I am not accustomed to monkeypatching. This will be easy to change.